gensim/models/fasttext_corpusfile.pyx

```diff
@@ -0,0 +1,287 @@
+            subwords_idx[effective_words[0]] = word.subword_idx
+
+            if hs:
+                codelens[effective_words[0]] = word.code_len
+                codes[effective_words[0]] = word.code
+                points[effective_words[0]] = word.point
+
+            effective_words[0] += 1
+            if effective_words[0] == MAX_SENTENCE_LEN:
+                break
+
+        # keep track of which words go into which sentence, so we don't train
+        # across sentence boundaries.
+        # indices of sentence number X are between <sentence_idx[X], sentence_idx[X])
+        effective_sentences[0] += 1
+        sentence_idx[effective_sentences[0]] = effective_words[0]
+
+        if effective_words[0] == MAX_SENTENCE_LEN:
+            break
+
+    # precompute "reduced window" offsets in a single randint() call
+    for i in range(effective_words[0]):
+        reduced_windows[i] = random_int32(next_random) % window
+
+
+def train_epoch_sg(model, corpus_file, offset, _cython_vocab, _cur_epoch, _expected_examples, _expected_words, _work,
+                   _l1):
+    """Train Skipgram model for one epoch by training on an input stream. This function is used only in multistream mode.
+
+    Called internally from :meth:`~gensim.models.fasttext.FastText.train`.
+
+    Parameters
+    ----------
+    model : :class:`~gensim.models.fasttext.FastText`
+        The FastText model instance to train.
+    corpus_file : str
+        Path to corpus file.
+    _cur_epoch : int
+        Current epoch number. Used for calculating and decaying learning rate.
+    _work : np.ndarray
+        Private working memory for each worker.
+    _l1 : np.ndarray
+        Private working memory for each worker.
+
+    Returns
+    -------
+    int
+        Number of words in the vocabulary actually used for training (They already existed in the vocabulary
+        and were not discarded by negative sampling).
+    """
+    cdef FastTextConfig c
+
+    # For learning rate updates
+    cdef int cur_epoch = _cur_epoch
+    cdef int num_epochs = model.epochs
+    cdef int expected_examples = (-1 if _expected_examples is None else _expected_examples)
+    cdef int expected_words = (-1 if _expected_words is None else _expected_words)
+    cdef REAL_t start_alpha = model.alpha
+    cdef REAL_t end_alpha = model.min_alpha
+    cdef REAL_t _alpha = get_alpha(model.alpha, end_alpha, cur_epoch, num_epochs)
+
+    cdef CythonLineSentence input_stream = CythonLineSentence(corpus_file, offset)
+    cdef CythonVocab vocab = _cython_vocab
+
+    cdef int i, j, k
+    cdef int effective_words = 0, effective_sentences = 0
+    cdef int total_effective_words = 0, total_sentences = 0, total_words = 0
+    cdef int sent_idx, idx_start, idx_end
+
+    init_ft_config(&c, model, _alpha, _work, _l1)
+
+    # for preparing batches & training
+    cdef vector[vector[string]] sentences
+    cdef unsigned long long random_number
```

Comment:
again looks like `random_number` and `word` unused
please check that you have no unused variables in cython files (`flake8` don't check `.pyx` in current repo, worth to fix, but up to you)

Reply:
done